rules.js

```diff
@@ -10,50 +10,89 @@ import {
   // semantic classes
-  [/^(button|f)(-(.+))?$/, ([_]) => emitWarning(_, TYPES.removed)],
+  [/^(button|f)(-.+)?$/, ([_]) => emitWarning(_, TYPES.removed)],
   [/^input(--is-invalid|--is-disabled|--is-select__wrap|--is-read-only|-toggle)?$/, ([_]) => emitWarning(_, TYPES.removed)],
-  [/^segment-control((-|_).+)?$/,([_]) => emitWarning(_, TYPES.removed)],
-  [/^field(-hint|-label|)?$/,([_]) => emitWarning(_, TYPES.removed)],
+  [/^segment-control([-_].+)?$/,([_]) => emitWarning(_, TYPES.removed)],
+  [/^field(-hint|-label)?$/,([_]) => emitWarning(_, TYPES.removed)],
   [/^link(--(block|dark))?$/,([_]) => emitWarning(_, TYPES.removed)],
   // old design system classes (troika)
-  [/^t-(.+)$/, ([_]) => emitWarning(_, TYPES.removed)],
+  [/^t-.+$/, ([_]) => emitWarning(_, TYPES.removed)],
   // Do not target page-container, but container class with pseudo variants
   [/^([^page-].+)?container$/, ([_]) => emitWarning('container', TYPES.removed)],
   // miscellaneous
   [/^transition-gpu$/, ([_]) => emitWarning(_, TYPES.removed, "use 'transform-gpu' or 'will-change-*'")],
   [/^fixed-ios-fix$/,([_]) => emitWarning(_, TYPES.removed, "use 'transform translate-z-0'")],
   [/^focus-ring$/,  ([_]) => emitWarning(_, TYPES.replaced, "use 'focusable'")],
-  // deprecated Warp classes
-  [/^s-(.+)-default$/, ([_]) => emitWarning(_, TYPES.deprecated, `use '${_.replace('-default', '')}'`)],
-  [/^s-text-link-hover(-active)?$/, ([_]) => emitWarning(_, TYPES.deprecated, `use 's-text-link'`)],
-  [/^s-(.+)-active-hover$/, ([_]) => emitWarning(_, TYPES.deprecated, `use classes with '-selected-hover' if available'`)],
-  [/^color-focused?$/, ([_]) => emitWarning(_, TYPES.deprecated, "use 's-focused'")],
-  [/^color-background(-.+)?$/, ([_]) => emitWarning(_, TYPES.deprecated, COLOR_MESSAGES.background)],
-  [/^color-text(-.+)?$/, ([_]) => emitWarning(_, TYPES.deprecated, COLOR_MESSAGES.text)],
+  [/^color-focused?$/, ([_]) => emitWarning(_, TYPES.removed)],
+  [/^color-background(-.+)?$/, ([_]) => emitWarning(_, TYPES.removed, COLOR_MESSAGES.background)],
+  [/^color-text(-.+)?$/, ([_]) => emitWarning(_, TYPES.removed, COLOR_MESSAGES.text)],
+
+  // *** Warp v1 -> v2 ***
+  // replaced semantic classes
+  [/^(s-.+)-default$/, ([_, c]) => emitWarning(_, TYPES.replaced, `use '${c}'`)],
+  // removed semantic classes
+  [/^s-text-link-hover(-active)?$/, ([_]) => emitWarning(_, TYPES.removed, `use 's-text-link'`)],
+  [/^(s-.+)-active-hover$/, ([_, c]) => emitWarning(_, TYPES.removed, `use '${c}-selected-hover' (if available)'`)],
+  // removed internal classes
+  [/^i-(bg|border|text|shadow)-\$(.*)$/, ([_, prop, token]) => emitWarning(_, TYPES.removed, `all internal classes (classes with 'i-' prefix) and most component specific tokens have been removed, ${COLOR_MESSAGES[prop==='bg'?'background':prop]} (or use class with arbitrary token value instead: '${prop}-[--w-${token}]', if this particular token still exists)`)],
```

Comment:
There's one group of component classes that might benefit from a more custom message. It's `i-(bg|border)-badge-` classes. If someone uses those, we could advice using the Badge component, like we noted in the css repo here: https://github.com/warp-ds/css/blob/next/component-classes/index.js#L8

Reply:
If someone is using the Warp internal `i-` classes I believe they are deliberately not using our components. 😆 So not sure if that would help...

Reply:
I'll try to clarify what I meant 😄  If someone is using [ribbon classes](https://github.com/warp-ds/css/blob/next/component-classes/index.js#L9-L20) that we will remove from the @warp-ds/css/component-classes/index.js in v2, then those internal classes would be found by the preset.

Reply:
If they are using the internal classes for the since long deprecated Ribbon, then they most likely also will be importing the ribbon component classes... which will fail in v2. So I'm not really sure what message would help in this case? 🤷